state-chain/pallets/cf-lp/src/lib.rs

```diff
@@ -294,6 +301,7 @@ pub mod pallet {
 		pub fn deregister_lp_account(who: OriginFor<T>) -> DispatchResult {
 			const STABLE_ASSET: Asset = Asset::Usdc;
 			let account_id = T::AccountRoleRegistry::ensure_liquidity_provider(who)?;
+			T::PoolApi::sweep(&account_id)?;
```

Comment:
why do we sweep here? I mean, i understand that we don't want to deregister if they have orders, but why not make them do this explicitly?

Reply:
wdym? as far as I understood sweep doesn't close any order, it collects just the fees.
If they have open orders the extrinsic will fail on the next check (when the other PR updating `open_orders_count` to include zero sized orders will be merged)